### PR TITLE
FT2023-304: Created a config form and checked the validation.

### PR DESCRIPTION
**Pull request by kumareshinnoraft**
Task 3:

Created a config form which will have the following fields

1. Full Name
2. Phone Number
3. Email ID
4. Radio button to select Gender.
5. Submit button

Added validation for the following -

Phone Number:

1. Phone number
2. This will accept only numbers
3. 10 digits Indian number is only allowed.

Email:

1. The email should be in RFC format.
2. Check if the email is from the public domain(like Yahoo, Gmail, Outlook etc or not).
3. blacklist any email id whose extension is not .com

### CODE REVIEW

File: web/modules/custom/config_form/config_form.routing.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+config_form.admin_api_form:
+  path: '/admin/config-form'
+  defaults:
+    _title: 'Fill the config form'
+    _form: 'Drupal\config_form\Form\ConfigForm'
+  requirements:
+    _permission: 'administer'
```
Comment (acheryyamanisha) [inline web/modules/custom/config_form/config_form.routing.yml:7]: does this work?

File: web/modules/custom/config_form/src/Form/ConfigForm.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    $form['gender'] = [
+      '#type'    => 'radios',
+      '#title'   => $this->t('Gender'),
+      '#options' => [
+        'male'   => $this->t('Male'),
+        'female' => $this->t('Female'),
+        'others' => $this->t('Other'),
+      ],
+      '#required' => TRUE,
+    ];
```
Comment (acheryyamanisha) [inline web/modules/custom/config_form/src/Form/ConfigForm.php:78]: Does the form store the default value once saved?

File: web/modules/custom/config_form/src/Form/ConfigForm.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+   * @param FormStateInterface 
+   *   Holds the values of the input data.
+   * 
+   * @return void
+   *   This function is validating the input data.
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state)
+  {
+    $phone_number = $form_state->getValue('phone_number');
+    $email        = $form_state->getValue('email');
```
Comment (acheryyamanisha) [inline web/modules/custom/config_form/src/Form/ConfigForm.php:102]: The logic is correct.
Can we achieve the same using a hook? Considering the form is not a custom one.

**Review by acheryyamanisha — CHANGES_REQUESTED** [review]
Fix the coding standards wherever applicable.

File: web/modules/custom/config_form/src/Form/ConfigForm.php (hunk 3 above)
Comment (kumareshinnoraft) [reply web/modules/custom/config_form/src/Form/ConfigForm.php:102]: Yes, it is possible using hook_form_alter() hook.

File: web/modules/custom/config_form/config_form.routing.yml (hunk 1 above)
Comment (kumareshinnoraft) [reply web/modules/custom/config_form/config_form.routing.yml:7]: Yes, ma'am this `administer` permission allow admin to access the page.

**Pull request merged**